### PR TITLE
modules/py_tf: Cleanup segment, detect, and callback data.

### PR DESCRIPTION
**Pull request by kwagyeman**
Addresses everything but the two invoke issues here (https://github.com/openmv/openmv/issues/2024). I'll send a PR to fix that next with a libtf update.

### CODE REVIEW

File: src/omv/modules/py_tf.c [hunk 1]
```diff
@@ -655,7 +691,7 @@ STATIC mp_obj_t int_py_tf_segment(bool detecting_mode, uint n_args, const mp_obj
     py_tf_input_data_callback_data.img = image;
     py_tf_input_data_callback_data.roi = &roi;
 
-    py_tf_segment_output_data_callback_data_t py_tf_segment_output_data_callback_data;
+    mp_obj_t py_tf_segment_output_data_callback_data;
```
Comment (iabdalkader) [inline src/omv/modules/py_tf.c:694]:
```suggestion
    mp_obj_t py_tf_detect_output_data_callback_data;
```

I know it's just a variable name, but it seems you want to match the function name.

File: src/omv/modules/py_tf.c [hunk 2]
```diff
@@ -1,8 +1,8 @@
 /*
  * This file is part of the OpenMV project.
  *
- * Copyright (c) 2013-2021 Ibrahim Abdelkader <iabdalkader@openmv.io>
- * Copyright (c) 2013-2021 Kwabena W. Agyeman <kwagyeman@openmv.io>
+ * Copyright (c) 2013-2023 Ibrahim Abdelkader <iabdalkader@openmv.io>
+ * Copyright (c) 2013-2023 Kwabena W. Agyeman <kwagyeman@openmv.io>
```
Comment (iabdalkader) [inline src/omv/modules/py_tf.c:5]:
```suggestion
 * Copyright (c) 2013-2024 Ibrahim Abdelkader <iabdalkader@openmv.io>
 * Copyright (c) 2013-2024 Kwabena W. Agyeman <kwagyeman@openmv.io>
```

File: src/omv/modules/py_tf.c [hunk 3]
```diff
@@ -629,7 +623,49 @@ STATIC void py_tf_segment_output_data_callback(void *callback_data,
...
+
+    mp_obj_t py_tf_segment_output_data_callback_data;
+
+    if (libtf_invoke(model->model_data,
+                     tensor_arena,
+                     &model->params,
+                     py_tf_input_data_callback,
+                     &py_tf_input_data_callback_data,
+                     py_tf_segment_output_data_callback,
+                     &py_tf_segment_output_data_callback_data) != 0) {
```
Comment (iabdalkader) [inline src/omv/modules/py_tf.c:657]: I think those variables/functions can be shortened out and made more readable, for example:

```suggestion
                     py_tf_input_callback,
                     &py_tf_input_callback_data,
                     py_tf_segment_output_callback,
                     &py_tf_segment_output_callback_data) != 0) {
```
And the same goes for classify.

**Review by iabdalkader — CHANGES_REQUESTED** [review]

File: src/omv/modules/py_tf.c (hunk 1 above)
Comment (kwagyeman) [reply src/omv/modules/py_tf.c:694]: Done

File: src/omv/modules/py_tf.c (hunk 2 above)
Comment (kwagyeman) [reply src/omv/modules/py_tf.c:5]: Done

File: src/omv/modules/py_tf.c (hunk 1 above)
Comment (kwagyeman) [reply src/omv/modules/py_tf.c:694]: Wait, for the following suggestion, I just renamed everything to remove the data piece. This should match the callback name. So, it shouldn't be detect as the callback name is segment.

File: src/omv/modules/py_tf.c (hunk 3 above)
Comment (kwagyeman) [reply src/omv/modules/py_tf.c:657]: Done

Comment (iabdalkader) [reply src/omv/modules/py_tf.c:657]: Did you forgot to push the changes ?

Comment (kwagyeman) [reply src/omv/modules/py_tf.c:657]: About to.

**Comment by kwagyeman** [comment]
@iabdalkader Updated

**Pull request merged**